Source/WebDriver/socket/HTTPServerSocket.cpp

```diff
@@ -108,7 +108,13 @@ void HTTPRequestHandler::sendResponse(HTTPRequestHandler::Response&& response)
 {
     auto& endpoint = RemoteInspectorSocketEndpoint::singleton();
     auto packet = packHTTPMessage(WTFMove(response));
-    endpoint.send(m_client.value(), packet.utf8().data(), packet.length());
+
+    auto result = packet.tryGetUTF8([&](std::span<const char> span) -> bool {
+        endpoint.send(m_client.value(), reinterpret_cast<const uint8_t*>(span.data()), span.size());
+        return true;
+    });
+    RELEASE_ASSERT(result);
```

Comment:
If we know in advance that getting UTF-8 is going to be successful, is there some reason to not just use `packet.utf8()`?

Secondly but perhaps more importantly, is this the right layer for making this fix? I am not familiar with this code, but I'm surprised that `RemoteInspectorSocketEndpoint::send` would take a `uint8_t` pointer and a size and then try to interpret that as latin-1. That feels like it's potentially a bug in `RemoteInspectorSocketEndpoint`, not here.

Reply:
Thank you for the review.

WTFString created by `String::fromUTF8()` keeps the internal data in `m_data16`, not `m_data8`.
In this case, the `.utf8()` will not return the correct result.
The reason for using `tryGetUTF8()` is to look at the result of `is8Bit()` and return the correct result for both `m_data8` and `m_data16`.

There may be some misunderstanding about the latter part.
`RemoteInspectorSocketEndpoint` is, as the name implies, the endpoint of a socket, so it is not responsible for its contents.
The side receiving the data via the socket, in this case the WebDriver user, receives the data as UTF-8 HTML.
Therefore, I don't see any particular problem in sending `uint8_t*` data obtained by `tryGetUTF8()`.

Reply:
I'm sorry, I made a mistake in my previous comment. 
Your review made me aware of it.

The data given to send was fine with `utf8()`.
I checked `utf8()` and `tryGetUTF8()` again and both returned the same data.
It was the length that was wrong.
I should have used `packet.utf8().length()` indices the length of the encoded string, instead of `packet.length()` indices the length of the original string.